src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Palaso.Reporting;
+using Palaso.UI.WindowsForms.ImageToolbox;
+
+namespace Bloom.ImageProcessing
+{
+	/// <summary>
+	/// Gecko struggles with hi-res images intented for printing. Gecko chews up memory, makes for slow drawing,
+	/// or even gives up on displaying the image altogether (worse on slow machines).
+	/// This cache takes requests for images and returns lo-res versions of them.
+	/// </summary>
+	public class RuntimeImageProcessor : IDisposable
+	{
+		private readonly BookRenamedEvent _bookRenamedEvent;
+		public int TargetDimension = 500;
+		private Dictionary<string, string> _originalPathToProcessedVersionPath;
+		private string _cacheFolder;
+
+		private ImageAttributes _transparentImageAttributes;
+
+		public RuntimeImageProcessor(BookRenamedEvent bookRenamedEvent)
+		{
+			_bookRenamedEvent = bookRenamedEvent;
+			_originalPathToProcessedVersionPath = new Dictionary<string, string>();
+			_cacheFolder = Path.Combine(Path.GetTempPath(), "Bloom");
+			_bookRenamedEvent.Subscribe(OnBookRenamed);
+			_transparentImageAttributes = new ImageAttributes();
+			_transparentImageAttributes.SetColorKey(Color.FromArgb(253, 253, 253), Color.White);
+		}
+
+		private void OnBookRenamed(KeyValuePair<string, string> fromPathAndToPath)
+		{
+			//Note, we don't pay attention to what the change was, we just purge the whole cache
+
+			TryToDeleteCachedImages();
+			_originalPathToProcessedVersionPath = new Dictionary<string, string>();
+		}
+
+		public void Dispose()
+		{
+			if (_originalPathToProcessedVersionPath == null)
+				return;
+
+			TryToDeleteCachedImages();
+			_originalPathToProcessedVersionPath = null;
+
+			//NB: this turns out to be dangerous. Without it, we still delete all we can, leave some files around
+			//each time, and then deleting them on the next run
+			//			_cacheFolder.Dispose();
```

Comment:
This probably isn't a real change, but I can't tell what is changed so looking at all. Seems it would be helpful to describe the danger of disposing the cache folder.